File: 02-後端/fastapi/09-ai/02-ollamaPython/ollama.py
import requests
import json

def chat(question):
    payload = {
        "model": "llama3.2:3b",
        "prompt": question
    }

    response = requests.post("http://localhost:11434/api/generate", json=payload, timeout=10000)
    # print(f'response.text={response.text}')

    items = response.text.strip().split('\n')
    # print('items=', items)

    tokens = []
    for item in items:
        # print('item=', item)
        o = json.loads(item)
        tokens.append(o['response'])

    response_text = ' '.join(tokens)
    return response_text

if __name__ == '__main__':
    print(chat("法國的首都在哪裡？"))

File: _more/A8-人工智慧/02-ollamaChat/ollama.js
const url = 'http://localhost:11434/api/chat';

export async function chat(q) {
    const data = {
        model: "llama3.2:3b", // "gemma:2b",
        messages: [
            { role: "user", content: q }
        ]
    };

    const response = await fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(data)
    });

    const text = await response.text();
    const lines = text.trim().split('\n');
    const json1 = '[' + lines.join(',\n') + ']';
    const jsonResponse = JSON.parse(json1);
    let r = []
    for (let t of jsonResponse) {
        r.push(t.message.content)
    }
    let responseText = r.join('')
    return responseText
}
